File: w1d1/two_sum.rb
# Finds a pair of numbers in an array that sum to zero

class Array
  def two_sum
    arr = self

    arr.each_with_index do |num, ind1|
      number_to_comp = num
      arr.each_with_index do |item, ind2|
        return true if number_to_comp+item == 0 && (ind1 != ind2)
      end
    end
    false
  end
end

puts [0,1,2,3].two_sum
puts [-1,1,2,3].two_sum

File: w6d6/two_sum.js
function twoSum(array) {
	var resArray = [];
	for(var i = 0; i < array.length - 1; i++) {
		for(var j = i + 1; j < array.length; j++) {
			if ((array[i] + array[j]) === 0) {
				var pair = [i, j];
				resArray.push(pair);
			}
		}
	}
	
	return resArray;
}

console.log(twoSum([5,1,-7,-5]));
console.log(twoSum([5,-1,-5,1]));
console.log(twoSum([0, 1, 2, 0]));